### PR TITLE
Fixes and cleanup in Secure Pairing class

### PR DESCRIPTION
**Pull request by pan-apple**
#### Problem
Some interface changes and bug fixes were identified during integration of secure pairing session. Need those changes in the code.

 #### Summary of Changes
This change modifies the class interface based on CHIP controller and Rendezvous Session integration. Also, there's cleanup in memory management for handshake messages.

Changes in Controller and Rendezvous session will be part of a separate PR.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Comment by pan-apple** [comment]
tag: @saurabhst, @jelderton, @BroderickCarlin

**Review by woody-apple — APPROVED** [review]

**Comment by pan-apple** [comment]
@saurabhst, @jelderton, @BroderickCarlin
I'd appreciate any feedback on this PR.
I have a follow on PR that brings in secure pairing (but is dependent on the code in this PR).

**Comment by woody-apple** [comment]
@saurabhst @jelderton ?

File: src/transport/SecurePairingSession.cpp [hunk 1]
```diff
@@ -53,14 +54,17 @@ SecurePairingSession::~SecurePairingSession(void)
...
-                                      SecurePairingSessionDelegate * delegate)
+                                      Optional<NodeId> myNodeId, uint16_t myKeyId, SecurePairingSessionDelegate * delegate)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
 
     VerifyOrExit(salt != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrExit(saltLen > 0, err = CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrExit(delegate != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
 
+    ChipLogProgress(Inet, "Secure Pairing: SetupCode %d, Iter %d, SaltLen %ld", setupCode, pbkdf2IterCount, saltLen);
```
Comment (jelderton) [inline src/transport/SecurePairingSession.cpp:65]: I'm wondering if we should be logging this information.  It seems to me like a potential security risk.

Comment (pan-apple) [reply src/transport/SecurePairingSession.cpp]: @jelderton , makes sense. I have remove the logs in the new commit.

**Review by saurabhst — APPROVED** [review]

**Pull request merged**